Markup.cpp

```diff
@@ -1163,35 +1146,38 @@ int _vscprintf(const char* format, va_list pargs)
 }
 #endif
 
-std::string CMarkup::Format(const char *fmt, ...)
+std::string CMarkup::Format(const char* fmt, ...)
 {
-    std::string retStr;
-    if (NULL != fmt)
+    if (!fmt)
     {
-        va_list marker;
+        return "";  // EARLY EXIT
+    }
 
-        // initialize variable arguments
-        va_start(marker, fmt);
+    va_list marker;
+    va_start(marker, fmt);
+    va_list markerCopy;  // Used for determining buffer size
+    va_copy(markerCopy, marker);
 
-        // Get formatted string length adding one for NULL
-        int len = _vscprintf(fmt, marker) + 1;
+    // Determine buffer size
+#if defined(_MSC_VER)
+    int len = _vscprintf(fmt, markerCopy) + 1;
+#else
+    int len = vsnprintf(nullptr, 0, fmt, markerCopy) + 1;
+#endif
+    va_end(markerCopy);
 
-        // Create a char vector to hold the formatted string.
-        std::vector<char> buffer(len, '\0');
-#ifdef _WIN32
-        int nWritten = _vsnprintf_s(&buffer[0], buffer.size(), len, fmt, marker);
+    std::vector<char> buffer(len);  // Create buffer of required size
+
+    // Format the string into the buffer
+#if defined(_MSC_VER)
```

Comment:
See if it's reasonable to replace this entire if-else statement with vsnprintf_s.

Reply:
Removed Windows-specific code.